doctor/types.py

```diff
@@ -93,6 +98,31 @@ def __new__(cls, *args, **kwargs):
             if not re.search(cls.pattern, value):
                 raise TypeSystemError(cls=cls, code='pattern')
 
+        # Validate format, if specified
```

Comment:
This logic is basically identical to the builtin formats that the python jsonschema library provided.